drivers/sensor/st/stmemsc/stmemsc_i2c.c

```diff
@@ -34,6 +39,10 @@ int stmemsc_i2c_read_incr(const struct i2c_dt_spec *stmemsc,
 int stmemsc_i2c_write_incr(const struct i2c_dt_spec *stmemsc,
 			   uint8_t reg_addr, uint8_t *value, uint8_t len)
 {
-	reg_addr |= STMEMSC_I2C_ADDR_AUTO_INCR;
-	return stmemsc_i2c_write(stmemsc, reg_addr, value, len);
+	uint8_t buf[16];
+
+	buf[0] = reg_addr | STMEMSC_I2C_ADDR_AUTO_INCR;
+	memcpy(&buf[1], value, len);
```

Comment:
Should check that you don't overrun the buffer

Reply:
would an `ASSERT` be better or an if with a return code here?

Reply:
Since the buffer is statically sized it would be a programming error in the sensor driver in my view, I'd assert in this case.